src/main/java/org/testng/internal/Parameters.java

```diff
@@ -182,62 +190,107 @@ else if (ITestResult.class.equals(c)) {
     return result;
   }
 
+  private static Object[] createParametersForConstructor(Constructor constructor,
```

Comment:
That would cause the translation from `Array` to `List` to be done in `org.testng.internal.Parameters#createInstantiationParameters`. So I thought its best to have this just as an array.